src/targets/gpu/mlir.cpp

```diff
@@ -960,6 +952,24 @@ void adjust_param_shapes(module& m, const std::vector<shape>& inputs)
     }
 }
 
+std::string dump_mlir(const module& m, const std::vector<shape>& inputs)
+{
+    module mm;
+    const_module_ref mr = &m;
+    if(not inputs.empty())
+    {
+        mm = m;
+        mr = &mm;
```

Comment:
Why does it require const_ref ? Shouldn't just copied module `mm` work ?

Reply:
I only want to copy the module if there is no input shapes because we wont be adjusting the parameters.

Reply:
Module with empty input shape is unlikely case it would have been const-folded.

Reply:
I think it is better to assume inputs are not empty. it would be simpler

Reply:
There is two overloads to `dump_mlir`. One just takes the module(which we dont want to copy) and the other overload takes the input shapes, which might be different than the shapes in the module so we need to modify the module which we will use a copy for this case.

Reply:
> and the other overload takes the input shapes, which might be different than the shapes in the module so we need to modify the module which we will use a copy for this case. 

Yes but those input shapes parameter would be input arguments to the precompile_op instruction. If they are empty that means MLIR module also doesn't take any inputs.

Reply:
When the inputs are empty it means skip doing param adjustments.